test/test_dependency_versions.py

```diff
@@ -61,9 +67,6 @@ def test_pinned_versions(tmp_path, python_version, build_frontend_env):
             pytest.skip("macOS arm64 does not support Python 3.6")
         constraint_filename = "constraints-python36.txt"
         build_pattern = "[cp]p36-*"
-    elif python_version == "3.7":
```

Comment:
Why is this removed?

Reply:
Because there is no parametrization on `python_version == "3.7"` and if there was one added, then the rest of the function does not account for it so I'd rather have either all or nothing and went with nothing.

Reply:
I reworked this differently in #1818, I'll resolve the conversation here. Let's discuss it in the new PR if need be.